v-next/hardhat-utils/src/errors/bytecode.ts

```diff
@@ -0,0 +1,48 @@
+import { CustomError } from "../error.js";
+
+export class InvalidLibraryAddressError extends CustomError {
+  constructor(formattedLibraries: string) {
+    super(`The following libraries have invalid addresses:
+${formattedLibraries}
+
+Please provide valid Ethereum addresses for these libraries.`);
+  }
+}
+
+export class AmbiguousLibraryNameError extends CustomError {
+  constructor(formattedLibraries: string) {
+    super(`The following libraries may resolve to multiple libraries:
+${formattedLibraries}
+
+Please provide the fully qualified name for these libraries.`);
+  }
+}
+
+export class UnnecessaryLibraryError extends CustomError {
+  constructor(formattedLibraries: string) {
+    super(`The following libraries are not referenced by the contract:
+${formattedLibraries}
+
+Please provide only the libraries that are actually needed.`);
```

Comment:
```suggestion
Please provide only the libraries that are needed.`);
```
Stop being mean to our poor users.